homeassistant/helpers/device_registry.py

```diff
@@ -608,7 +614,7 @@ def _substitute_name_placeholders(
         try:
             return name.format(**translation_placeholders)
         except KeyError as err:
-            if get_release_channel() != "stable":
+            if get_release_channel() != ReleaseChannel.STABLE:
```

Comment:
But in these cases there are enums at both sides right?

Reply:
Right. I don't see a case where it would be a str there.

Reply:
Yes both sides are enums that can also behave as strings if needed but from a linter and type checking perspective, an enum is a bounded string (with certain values allowed) while a string could have any kind of data inside.